wordpress-mobile/WordPress-Android
# [Tags IA] Fetch posts for multiple tags

Fixes #20588 

Move shared fetching logic to Repository and create ViewModel to use Repository to fetch posts for tags directly.

-----

## To Test:

- Open Jetpack
- Enable the `reader_tags_feed` feature flag in Debug Settings
- Go to Reader
- Select the `Tags` feed
- **Verify** a UI shows up that loads the tag posts for the followed tags

-----

## Regression Notes

1. Potential unintended areas of impact

    - Fetching posts in other places of the Reader

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual tests
    - Unit tests of Repository
    - Unit tests of existing code

3. What automated tests I added (or what prevented me from doing so)

    - Unit tests in all new classes introduced

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
N/A, the UI in this PR is completely throwaway and only there to easily test these changes.
